bin/reth/src/chain/import.rs

```diff
@@ -0,0 +1,197 @@
+    /// - macOS: `$HOME/Library/Application Support/reth/db`
+    #[arg(long, value_name = "PATH", verbatim_doc_comment, default_value_t)]
+    db: PlatformPath<DbPath>,
+
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    ///
+    /// Built-in chains:
+    /// - mainnet
+    /// - goerli
+    /// - sepolia
+    #[arg(
+        long,
+        value_name = "CHAIN_OR_PATH",
+        verbatim_doc_comment,
+        default_value = "mainnet",
+        value_parser = genesis_value_parser
+    )]
+    chain: ChainSpec,
+
+    /// The block file to import.
+    #[arg(long, value_name = "IMPORT_PATH", verbatim_doc_comment)]
+    path: PlatformPath<ConfigPath>,
+}
+
+impl ImportCommand {
+    /// Execute the `import` command
+    pub async fn execute(&self) -> eyre::Result<()> {
+        info!(target: "reth::cli", "reth import starting");
+
+        debug!(target: "reth::cli", chain_spec = ?self.chain, "Importing chain with chainspec");
+
+        let config: Config = self.load_config()?;
+        info!(target: "reth::cli", path = %self.db, "Configuration loaded");
+
+        info!(target: "reth::cli", path = %self.db, "Opening database");
+        let db = Arc::new(init_db(&self.db)?);
+        info!(target: "reth::cli", "Database opened");
+
+        info!(target: "reth::cli", ttd=?self.chain.paris_ttd, "Initializing genesis");
+        let genesis_hash = init_genesis(db.clone(), self.chain.clone())?;
+
+        if genesis_hash != self.chain.genesis_hash() {
+            // TODO: better error text
+            return Err(eyre::eyre!(
+                "Genesis hash mismatch: expected {}, got {}",
+                self.chain.genesis_hash(),
+                genesis_hash
+            ))
+        }
+
+        // create a new FileClient
+        info!(target: "reth::cli", "Importing chain file");
+        let file_client = Arc::new(FileClient::new(&self.path).await?);
+        let tip = file_client.tip().expect("file client has no tip");
+        info!(target: "reth::cli", "Chain file imported");
+
+        let (consensus, notifier) = BeaconConsensus::builder().build(self.chain.clone());
+        info!(target: "reth::cli", "Consensus engine initialized");
+
+        notifier.send(ForkchoiceState {
+            head_block_hash: tip,
+            safe_block_hash: tip,
+            finalized_block_hash: tip,
+        })?;
+
+        // construct downloaders and start pipeline
+        let mut pipeline =
+            self.build_pipeline(&config, &file_client, consensus.clone(), &db).await?;
+
+        tokio::spawn(handle_events(pipeline.events().map(Into::into)));
+
+        // Run pipeline
+        info!(target: "reth::cli", "Starting sync pipeline");
+        pipeline.run(db.clone()).await?;
+
+        info!(target: "reth::cli", "Successfully completed block import!");
+        Ok(())
+    }
+
+    async fn build_pipeline(
```

Comment:
Refactor all this into `reth node --import <file>` or avoid duplicating the pipeline construction logic

Reply:
good idea, will refactor